doc/how_to/streamlit_migration/chat.md

```diff
@@ -0,0 +1,145 @@
+# Create Chat Interfaces
+
+Both Streamlit and Panel provides special components to help you build conversational apps.
+
+| Streamlit            | Panel               | Description                            |
+| -------------------- | ------------------- | -------------------------------------- |
+| `st.chat_message`    | [`pn.chat.ChatEntry`](../../../examples/reference/chat/ChatEntry.ipynb) | Output a single chat message  |
+| `st_chat_input`      |  | Input a chat message |
+| `st.status`          | | Display output of long-running tasks in a container |
+|                      | [`pn.chat.ChatFeed`](../../../examples/reference/chat/ChatFeed.ipynb)  | Output multiple of chat messages         |
+|                      | [`pn.chat.ChatInterface`](../../../examples/reference/chat/ChatInterface.ipynb)  | High-level, easy to use chat interface |
+| `langchain.callbacks.StreamlitCallbackHandler` | [`pn.chat.PanelCallbackHandler`](../../../examples/reference/chat/ChatInterface.ipynb) | Display the thoughts and actions of a LangChain agent |
+
+The starting point for most Panel users would be the *high-level*, easy to use `ChatInterface` and `PanelCallbackHandler` components. Not the lower level `ChatEntry` and `ChatFeed` components.
+
+You can find specialized chat components and examples at [panel-chat-examples/](https://holoviz-topics.github.io/panel-chat-examples/).
+
+## Chat Message
+
+Lets see how-to migrate an app using `st.chat_message`
+
+### Streamlit Chat Message Example
+
+```python
+import streamlit as st
+
+with st.chat_message("user"):
+    st.image("https://streamlit.io/images/brand/streamlit-logo-primary-colormark-darktext.png")
+    st.write("# A faster way to build and share data apps")
+```
+
+![Streamlit chat_entry](../../_static/images/streamlit_chat_message.png)
+
+### Panel Chat Message Example
+
+```python
+import panel as pn
+
+pn.extension(design="material")
+
+message = pn.Column(
+    "https://panel.holoviz.org/_images/logo_horizontal_light_theme.png",
+    "# The powerful data exploration & web app framework for Python"
+)
+pn.chat.ChatEntry(value=message, user="user").servable()
+```
+
+![Panel ChatEntry](../../_static/images/panel_chat_entry.png)
+
+## Chat Input
+
+### Streamlit Chat Input
+
+```python
+import streamlit as st
+
+prompt = st.chat_input("Say something")
+if prompt:
+    st.write(f"User has sent the following prompt: {prompt}")
+```
+
+![Streamlit chat_input](../../_static/images/streamlit_chat_input.png)
+
+### Panel Chat Input
+
+Panel does not provide a dedicated *chat input* component because it is built into Panels high-level `ChatInterface`.
+
+Below we will show you how to build a custom `ChatInput` widget.
+
+```python
```

Comment:
This custom `Viewer` component also shows some of the challenges it has.

- The component is not and cannot be used as a widget. I.e. you need to `pn.bind` to and `pn.depends` on `chat_input.param.value` not `chat_input`. There is no `.from_param` method etc.
- It takes a lot of boiler plat code to separate the *viewer* `params` from the *layout* params. And getting `sizing_mode`, `width`, ... properly linked up to the `_layout` is almost impossible.

I.e. you can make custom components using `Viewer`. But I can't see how you can create something that Panel would be willing to include as a widget.